cf-nodejs-logging-support-core/log-core.js

```diff
@@ -292,9 +294,15 @@ var initLog = function () {
     var logObject = JSON.parse(initDummy);
 
     var now = new Date();
-
     logObject.written_at = now.toJSON();
-    logObject.written_ts = now.getTime() * NS_PER_MS + (process.hrtime()[1] % 1000000);
+
+    var lower = process.hrtime()[1] % 1000000
+    var higher = now.getTime() * NS_PER_MS
+
+    logObject.written_ts = higher + lower;
+    if(logObject.written_ts < lastTimestamp)
+        logObject.written_ts += 1000000;
```

Comment:
This potentially advances time very fast, if there was another reason than the lower overflow. Also it is not clear from the code, that the result is actually larger than `lastTimestamp`. Can you add some comment on why this approach is taken. Naming conventions could also help here.

Reply:
rewrote the tests to be deterministic to test failures due to random chance and "bad" timings